metrics/cli/test/cli/command/update_command_test.dart

```diff
@@ -0,0 +1,189 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:args/args.dart';
+import 'package:cli/cli/command/doctor_command.dart';
+import 'package:cli/cli/command/update_command.dart';
+import 'package:cli/cli/updater/config/factory/update_config_factory.dart';
+import 'package:cli/cli/updater/factory/updater_factory.dart';
+import 'package:cli/cli/updater/updater.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../test_utils/matchers.dart';
+import '../../test_utils/mocks/update_config_mock.dart';
+
+void main() {
+  group("UpdateCommand", () {
+    const configPath = 'configPath';
+
+    final updaterFactory = _UpdaterFactoryMock();
+    final updateConfigFactory = _UpdateConfigFactoryMock();
+    final updater = _UpdaterMock();
+    final updateCommand = _UpdateCommandStub(
+      updaterFactory,
+      updateConfigFactory,
+      configPath: configPath,
+    );
+
+    PostExpectation<Updater> whenCreateUpdater() {
+      return when(updaterFactory.create());
+    }
+
+    tearDown(() {
+      reset(updaterFactory);
+      reset(updateConfigFactory);
+      reset(updater);
+    });
+
+    test(
+      "throws an ArgumentError if the given updater factory is null",
+      () {
+        expect(
+          () => DoctorCommand(null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".name equals to the 'update'",
+      () {
+        expect(updateCommand.name, equals('update'));
+      },
+    );
+
+    test(
+      ".description is not empty",
+      () {
+        final description = updateCommand.description;
+
+        expect(description, isNotEmpty);
+      },
+    );
+
+    test(
+      "has the 'config-file' option",
+      () {
+        final argParser = updateCommand.argParser;
+
+        expect(argParser.options, contains('config-file'));
+      },
+    );
+
+    test(
+      "'config-file' option has a non-empty help",
+      () {
+        final argParser = updateCommand.argParser;
+
+        final option = argParser.options['config-file'];
+
+        expect(option.help, isNotEmpty);
+      },
+    );
+
+    test(
+      "'config-file' option has a non-empty value help",
+      () {
+        final argParser = updateCommand.argParser;
+
+        final option = argParser.options['config-file'];
+
+        expect(option.valueHelp, isNotEmpty);
+      },
+    );
+
+    test(
+      ".run() creates updater using the given updater factory",
+      () async {
+        whenCreateUpdater().thenReturn(updater);
```

Comment:
✅ 
Maybe we should rethink this a little bit. I believe if our class has always the one and only one return object - preprogrammed - then we should create a stub implementation instead of mocking an object. For the class where we really need the mock implementation - we can create the separate Updater and use it to check with `.called(once)`.